readthedocs/api/v2/views/model_views.py

```diff
@@ -289,9 +290,26 @@ class BuildViewSetBase(UserSelectViewSet):
     )
     def running(self, request, **kwargs):
         project_slug = request.GET.get('project__slug')
+        query = Q(project__slug=project_slug)
+
+        project = get_object_or_404(Project, slug=project_slug)
+        # Limit concurrency for translations as a whole (projects using the
+        # same repository for all their translations projects will trigger a
+        # build per translation on each commit)
+        if project.main_language_project:
+            query = (
+                Q(project__main_language_project=project.main_language_project) |
+                Q(project=project.main_language_project)
+            )
+        elif project.translations.count():
+            query = (
+                Q(project__in=project.translations.all()) |
+                Q(project__slug=project_slug)
+            )
+
         queryset = (
             self.get_queryset()
-            .filter(project__slug=project_slug)
+            .filter(query)
             .exclude(state__in=[BUILD_STATE_TRIGGERED, BUILD_STATE_FINISHED])
         )
         return Response({'count': queryset.count()})
```

Comment:
I think we want to return a `project_count` and a `translation_count`, because we likely don't want to apply the same logic in both places.

Reply:
What would be those cases where we want that difference in logic?

Actually, I was thinking about simplifying this by just moving all the logic to a QuerySet and return just `True` or `False` depending if the limit is reached or not, instead of comparing all of this here and in the `trigger_build` function as well.

So, just `Project.objects.reached_max_concurrency(project)` -> `True` or `False` and we use this in the endpoint and in the `trigger_build` function. This also helps us to override it from corporate to have a different behavior. Similar to what we do with `Project.objects.is_active(project)`.

Reply:
I mentioned above, but 50 builds across all translations limited at 2 concurrency is a terrible experience. We likely want to be able to change the logic of how we handle translations vs. project builds.